### PR TITLE
Generate random password when master_password is empty

### PR DESCRIPTION
**Pull request by dkossako**
## what
* If master_password is not provided or empty random value will be used

## why
* AWS API requires to provide password
* Secrets should not be kept in repository

### CODE REVIEW

**Comment by dkossako** [comment]
can someone let me know what's wrong with bridgecrew analysis? I don't have access there

File: docs/terraform.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 | Name | Version |
 |------|---------|
 | <a name="provider_aws"></a> [aws](#provider\_aws) | >= 2.0 |
+| <a name="provider_random"></a> [random](#provider\_random) | n/a |
```
Comment (Gowiem) [inline docs/terraform.md:15]: @dkossako can you please add the random provider explicitly to `versions.tf`?

Comment (dkossako) [reply docs/terraform.md:15]: fixed

File: versions.tf [hunk 2]
```diff
@@ -10,5 +10,9 @@ terraform {
       source  = "hashicorp/local"
       version = ">= 1.3"
     }
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.1"
```
Comment (Gowiem) [inline versions.tf:15]: Apologies on this front as well, but can you downgrade this to the earliest random provider that will support the usage that you've added? We don't want to force module consumers to upgrade for no reason if they're pinning an earlier version of the random provider.

**Review by Gowiem — DISMISSED** [review]
2 other requests @dkossako -- Those will be my last. Thanks!

File: versions.tf (hunk 2 above)
Comment (dkossako) [reply versions.tf:15]: Sure, understand

File: main.tf [hunk 3]
```diff
@@ -39,11 +39,17 @@ resource "aws_security_group_rule" "ingress_cidr_blocks" {
   security_group_id = join("", aws_security_group.default.*.id)
 }
 
+resource "random_password" "password" {
+  count   = var.master_password != "" ? 0 : 1
```
[the remark this answers is not in the document]
Comment (dkossako) [reply main.tf:43]: Sure

**Review by Gowiem — APPROVED** [review]

**Comment by Gowiem** [comment]
@dkossako Thanks for the contribution! Released as https://github.com/cloudposse/terraform-aws-documentdb-cluster/releases/tag/0.15.0

**Pull request merged**